transition_sampling/algo/aimless_shooting.py

```diff
@@ -28,84 +31,241 @@ class AimlessShooting:
 
     Attributes
     ----------
-    TODO fill this out
+    engine : AbstractEngine
+        Engine used to run the simulations
+    position_dir : str
+        Directory containing xyz guesses of transition states.
+    results_dir : str
+        Directory where generated transition states will be stored.
+    current_offset : int
+        -1, 0 or +1. The delta-t offset the point being run with engine
+        currently represents for purposes of choosing the next point.
+    current_start : np.ndarray
+        xyz array of the current starting position. Has shape (num_atoms, 3)
+    accepted_states : list[np.ndarray]
+        A list of the accepted positions. Contains duplicates.
     """
     def __init__(self, engine: AbstractEngine, position_dir: str,
-                 results_dir: str, starting_xyz: str):
+                 results_dir: str):
         self.engine = engine
         self.position_dir = position_dir
         self.results_dir = results_dir
 
-        self.current_offset = 0
+        self.current_offset = random.choice([-1, 0, 1])
+        self.current_start = None
 
-        # This is a temporary work around while we get the starting directory
-        # working
-        if starting_xyz is not None:
-            with open(starting_xyz, "r") as file:
+        self.accepted_states = []
+
+    def run(self, n_points: int, n_state_tries: int, n_vel_tries: int) -> None:
+        """Run the aimless shooting algorithm to generate n_points.
+
+        Each state that is generated is written as a .xyz to the given results
+        directory.
+
+        There are two loops of tries before failing to generate a new transition
+        state. The outer loop is over starting positions and the inner loop is
+        over resampled velocities.
+
+        This implementation works as described below:
+        1. Pick a starting state
+        2. For that starting state, try n_vel_tries number of times to get it
+            accepted by regenerating the velocities each time it is rejected.
+        3. Either:
+            a. An accepted state was found. Pick the next starting point from
+                this accepted state and repeat from 1.
+            b. An accepted state was not found. Randomly select a state we know
+                has worked before and repeat from 1.
+        4. If 3b occurs n_state_tries in a row without generating a new accepted
+            state, we've failed. Raise an exception.
+
+        Parameters
+        ----------
+        n_points
+            Number of total points to generate
+        n_state_tries
+            Number of retries to find another point before failing
+        n_vel_tries
+            Number of retries to find another point before failing
```

Comment:
I think this would be better described as the number of resampled velocities on a single point before failing, or something to that effect.

Reply:
You are correct, I just copy pasted that from the old doc and forgot about it lol